src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -96,10 +99,11 @@ private async Task<bool> IsEnabledWithVariantsAsync<TContext>(string feature, TC
             {
                 isFeatureEnabled = false;
             }
-            else if ((featureDefinition.Variants?.Any() ?? false) && featureDefinition.Allocation != null)
```

Comment:
I removed the else here, as a disabled flag should still have a variant (the disabled default variant)

Reply:
Is this a fix on variants? or is it related to telemetry? If it's a fix on variants can you isolate it in a separate PR?

Reply:
It's only relevant to telemetry. For just variants, a call to IsEnabledAsync for a disabled flag doesn't actually care about the variant because the variant can't override the IsEnabled state (for disabled flags). But for telemetry we still want to log what the variant allocation is / would be.

This means there's a refactor in our future combining IsEnabledAsync and GetVariant further, but that's for a different PR.